### PR TITLE
Create employees entity and relationships, also add spec work

### PR DESCRIPTION
**Pull request by kenanwarren**
# Spec
Screen Shot 2021-09-13 at 10 21 22 AM

# Types
Screen Shot 2021-09-13 at 10 21 10 AM

# Output Graph
Screen Shot 2021-09-13 at 10 22 36 AM

### CODE REVIEW

File: .gitleaks.toml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+description = "Square OAuth secret"
+regex = '''sq0csp-[0-9A-Za-z\\-_]{43}'''
+tags = ["key", "square"]
+[[rules]]
+description = "Twilio API key"
+regex = '''(?i)twilio(.{0,20})?['\"][0-9a-f]{32}['\"]'''
+tags = ["key", "twilio"]
+[allowlist]
+commits = [
+  "384fbec9944b8fb7c87e3e286856addfb441c5cb",
```
Comment (kenanwarren) [inline .gitleaks.toml:104]: This is the offending commit, the photo url tokens seem to have expired long ago but it is causing issues with the build.

File: docs/spec/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {
```
Comment (ndowmon) [inline docs/spec/src/types.ts:1]: You can upgrade `@jupiterone/integration-sdk-core@6.19.0` (and other packages) - then you can import these types and the `toImplementSpec` matcher.

**Comment by kenanwarren** [comment]
Also very closely related is Stefan's work: https://github.com/JupiterOne/graph-bamboohr/pull/25 I wasn't aware of the original ticket when the support ticket was created so there was some duplicate work.

**Review by aiwilliams — APPROVED** [review]

File: src/client.ts [hunk 3]
```diff
@@ -103,23 +102,36 @@ export class APIClient {
...
+        employeeDetails: employees.get(user.employeeId.toString()) || {},
       });
     }
   }
 
-  private async fetchEmployeeDirectory(): Promise<BambooHREmployeesMap> {
+  public async iterateEmployees(
+    iteratee: ResourceIteratee<BambooHREmployee>,
+  ): Promise<void> {
+    const employees = await this.fetchEmployeeDirectory();
```
Comment (aiwilliams) [inline src/client.ts:113]: I missed this earlier, but it looks like `await this.fetchEmployeeDirectory();` is being called during `iterateEmployees` and `iterateUsers`. What do you think about caching that data? Is the complete set of employees in a single request to `'v1/employees/directory'`?

**Pull request merged**